Source/engine/animationinfo.cpp

```diff
@@ -166,15 +166,18 @@ void AnimationInfo::ChangeAnimationData(const CelSprite *celSprite, int numberOf
 {
 	if (numberOfFrames != NumberOfFrames || ticksPerFrame != TicksPerFrame) {
 		// Ensure that the CurrentFrame is still valid and that we disable ADL cause the calculcated values (for example TickModifier) could be wrong
-		CurrentFrame = clamp(CurrentFrame, 1, numberOfFrames);
+		if (numberOfFrames >= 1)
+			CurrentFrame = clamp(CurrentFrame, 1, numberOfFrames);
+		else
+			CurrentFrame = 0;

```

Comment:
This is necessary because `clamp(int 1, 0)` is invalid, fails assertions in debug builds and is called out as undefined behaviour for std::clamp - https://en.cppreference.com/w/cpp/algorithm/clamp

Reply:
Would it make sense to add support for this in our own abstraction? We have a polyfill for `clamp` right now right?

Alternatively, would it be possible to map this particular logic to another operation name? Semantically, what are we doing here?

Reply:
It's a polyfil, so should act as much as the real thing as possible, eventually we should be able to just switch to the real thing without any changes needed

Reply:
> Alternatively, would it be possible to map this particular logic to another operation name? Semantically, what are we doing here?

I'm not familiar enough with the animation code to confidently say why this is the case, but for a animation sequence with 4 frames (0-3) when played "forwards" the animation sequence starts at frame 1 and goes to frame 3, before wrapping back to frame 1. For animations played "backwards" the animation sequence starts at frame 3 and goes to frame 0 before wrapping back around to frame 3.

https://github.com/diasurgical/devilutionX/blob/ec1850e2950f015d7814a8d9472bde471f4cb0d6/Source/engine/animationinfo.cpp#L188-L200

If all sequences started from 0 we could handle the case for numberOfFrames (really the last frame number) == 0 with the original code, `clamp(int, 0, 0)` is valid.